trikot-viewmodels-declarative-compiler/viewmodels-declarative-compiler-core/src/main/kotlin/com/mirego/trikot/viewmodels/declarative/compiler/BaseVMDCodeGenerator.kt

```diff
@@ -44,7 +45,11 @@ abstract class BaseVMDCodeGenerator(
     fun generateViewModelParents(viewModelMetaDataList: List<ViewModelMetaData>) {
         logger.logging("Generating abstract parents for viewmodels: $viewModelMetaDataList")
         viewModelMetaDataList.forEach {
-            generateAbstractParent(it)
+            try {
+                generateAbstractParent(it)
+            } catch (e: Exception) {
+                throw Exception("Unable to generate class with metadata = $it", e)
```

Comment:
Exception type seems a bit too wide for both the catching and rethrowing. Can't we catch only expected type (maybe IOException here?) 
Also for throwing either a custom exception type or a relevant one?

Reply:
Not sure about both... this exception isn't meant to be caught by anyone. Exceptions here are always "process terminating", and we catch / rethrow only for logging/debugging purpose. So there is not much value in either fine-grained catching or even throwing a more specific type. We just want to log whatever happened, no matter what it is.